components/constellation/constellation.rs

```diff
@@ -2141,6 +2135,35 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         }
     }
 
+    fn notify_history_changed(&self, frame_id: FrameId) {
+        if let Some(frame) = self.frames.get(&frame_id) {
```

Comment:
Can we not use one of the existing iterators for this? This appears to be only sending the session history for the frame that traversed, not the joint session history.